### ISSUE TITLE
Bug: Filtered Handler Extensions result in massively large files due to class instance variables being logged.

### ISSUE THREAD

**Issue by jspaleta** [sensu/sensu#1872]
Logging in filter.rb when using a handler extension can lead to very bad behavior when the extension uses class instance variables. The class instance variables are converted into json when the handler is passed into the logging function as data.
```
      def filter_event(handler, event)
        details = {:handler => handler, :event => event}
        ...
       if filter_message
          @logger.info(filter_message, details)
```
This can produce very very large logs very quickly. 
Tested by holding onto a logger instance as a class variable. The logging stream yet to be written to disk is encoded into json. Crazy!  User ran into this problem in #help exactly due to the logger variable.

This might even result in recursion if the handler extension has a looping child/parent relationship with other objects via class instance variables.  I haven't tested, but i'm very suspicious. Trying to figure out a way to test that safely.

## Expected Behavior
Handler extensions should be able to make use of class instance variables, without them leaking into log files, generating huge files.

## Current Behavior
handler as encoded into details hash and passed to logger will have its class instance variable serialized into json, resulting in huge log files.

## Possible Solution
Test to see if handler is an extension, if so then just pass a safer subset the handler extension info into the details hash as a hash. Perhaps just the handler extension definition method. 

## Steps to Reproduce (for bugs)
1. Build and handler extension with filters defined in its definition
2. Define a bad_logger instance variable in postint
3. install this extension and use it in a check.
4. filter will kick in and log the handler, including a json representation of @bad_logger

I'll attach a mock extension that has the necessary bits

## Context
Ran in to this trying to help a user debug custom extension. I eventually track down to the filter.rb logging action, but there doesn't seem to be a way to build an extension and hide those class instance variables from the json parser when it dumps the handler object to the log string.  

I think we need to make sure the filter logging is smarter about how it details handler extensions, as they aren't pure configuration hashes.

## PR #1873

**Comment by jspaleta** [comment]
[test_ext.rb.txt](https://github.com/sensu/sensu/files/2096281/test_ext.rb.txt)
Mock handler extension with inline comments

**Comment by portertech** [comment]
Closed by #1873 (it will be included in the next core release, patch level or greater)
